src/material-experimental/mdc-radio/harness/radio-harness-filters.ts

```diff
@@ -6,6 +6,11 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
+export type RadioGroupHarnessFilters = {
```

Comment:
It kind of feels like `id` and `name` should just be things you can do with normal selectors and shouldn't need a special filter option, e.g.

```ts
const myIdLoader = await loader.getChildLoader('#my-id')
const harness = myIdLoader.getHarness(RadioGroupHarness);
```

There is a problem with this currently though, which is that `getHarness` searches for the harness on an element _under_ the root of the loader. So if `my-id` is the id of your radio group, it won't check that element, just the ones under it. Maybe `getHarness` should check the given element and anything under it? That would eliminate the need for these filters I think

Reply:
For this case isn't it a bit more special though because the `name` or `id` is not necessarily set on the `radio-group` or `radio-button` element, but rather on the underlying input.

So if someone queries a radio-group by name, we check if it has a DOM attribute or find the right group by respecting the `name` attribute for child radio-buttons in groups.

Another pain point here is that if someone sets the radio-group name through an input binding, the `name` attribute will not be set. There is no host-binding for that. That's why need to respect the inputs of child radio-button's.

Reply:
Ah I see that's a good point, I wonder if I should add similar ones for checkbox. It will probably have the same issue if people want to search by name...

Reply:
Good point. I will double-check on that and make a note.